### PR TITLE
Client Registration/Login 

### PR DESCRIPTION
**Pull request by thomasdiaz1**
Hi Team,

As discussed, adding a pull-request with the current implementation of the user login/registration features on the client side.

Changes:
(+) React-Redux for state management. Wanted to store a user there so that once it does have necessary params, we could render content per that user.
(+) Service communicates to the server side features that tiffany built out. Confirming I able able to login as a user successfully, permitting that user exists. Confirming I am also able to register a new user from the front-end.
(+) Basic forms for login and register have been built out. You can also navigate between the login and register pages through hyperlinks

Let me know if you have questions. Later today I will be working on refactoring the register and login actions to reduce the number of api calls.

Thanks,
Tom

### CODE REVIEW

File: src/reducers/user-reducer.js
Comment (tchen007) [inline src/reducers/user-reducer.js:28]: NIT: I believe you can just return an empty user like this
```
return {
     user: {}
}
```

File: src/actions/user-actions.js
Comment (tchen007) [inline src/actions/user-actions.js:33]: This is not needed you can just call the login like below.  The login endpoint checks if the user is in the database. I would attempt to hit the endpoint with a user that does not exist and see what the response is. If it returns an error please use `catch()` to handle or if it returns something different use that logic to handle it. I would recommend fixing this before merging into main. Thanks and please let me know if you have any questions.

```
userService.login(user).then(response => dispatch({...
```

File: src/actions/user-actions.js
Comment (tchen007) [inline src/actions/user-actions.js:49]: Instead of this:
You can possibly use 
```
.catch(alert("Username...."))
```

File: src/actions/user-actions.js
Comment (tchen007) [inline src/actions/user-actions.js:9]: I don't believe a `flag` variable is really needed. Cleaner code would probably be something along the lines of 
```
if user exists: => alert and return or break
else:
createUser with service
```

**Review by tchen007 — CHANGES_REQUESTED** [review]
Overall solid work! The reducer and services look good to me, but I did not test them. I had some comments on user-actions and I will defer to @kris103 for the look and feel of `Login` and `Register`. Please take a look at the comments and make changes as needed or reply with your response. Let me know if you have any questions.

**Comment by thomasdiaz1** [comment]
@kris103 @tchen007 

Main has been merged into Tom-Client to reflect design changes in login/register components. Linked everything back up and confirmed that I was still able to login as a user from the front-end at the network level (no errors); I can also post new users with the params Kris put in-place.

Chose not to include credit card information as its not in the current model. Should we do that or is this just a step to validate a user's intention in becoming an Otaku? 

Also note I did a few test runs while doing some debugging and had incidentally created bad data points (this is now fixed in the code but the issue remains in the SQL db); userID 9 is completely blank; userID 7 is a dupe username of userID 5. Ideally both should not exist... worth us cleaning that up or no? The dupe username causes issues for logging in as 'tdiaz' but otherwise is not a problem at the moment.

Let me know if you require a new pull request given there were a number of changes over this one's lifecycle. As a recap:

(+) React-Redux for state management. Wanted to store a user there so that once it does have necessary params, we could render content per that user.
(+) Service communicates to the server side features that tiffany built out. Confirming I able able to login as a user successfully, permitting that user exists. Confirming I am also able to register a new user from the front-end.
(+) Login and Registration forms from main have been merged into the tom-client to fit Kris' design pattern. 


*Thing to note here, should we be handling on the front-end any failures from logging in or registering? Right now I have an alert thrown, but the user is still being redirected back to the home page. Thoughts on how to return maybe an error or success message that the react client can then field before redirecting the user the home? This is implemented both for the registration and login pages. It's not a big deal, but if there is user error it will result in more clicks as they will have to re-navigate to the appropriate page.

Thanks,
Tom

File: src/services/group-service.js
Comment (tchen007) [inline src/services/group-service.js:4]: This is on me, I changed the endpoint. For consistency and sanity I think we are trying to keep it `groups` on the client side. Can you please rename/refactor to `findAllGroups` and the endpoint will be:
```
/api/groups
```

Additionally for this file can you write accompanying service/reducer for creating a group. Please see the server repo's `ClubController.java` line 20 for the endpoint you will be connecting to.  Thanks!

**Review by tchen007 — CHANGES_REQUESTED** [review]
Hey Tom, 
Great work! The only reason why this PR is not getting approved for merge is because this has a lot of moving parts. But it's really close!  As for the failures from logging/registering, it shouldn't redirect to home? Shouldn't it redirect to its respective page? @kris103 You probably have the best knowledge about this. Can you look at his last point and work on it together?

Per your note about the DB, I deleted the following userIds: 7, 9. and 8 (no usertype). Please lmk if you see any other weird data and I can delete it to try to keep it clean!
Per your question,`Should we do that or is this just a step to validate a user's intention in becoming an Otaku?` - I believe this is more of a dummy data that goes nowhere so I don't think we necessarily need to validate except. Not 100% necessary but we could just make the fields required? But I'll leave it for you and @kris103 to decide! 👍 

Besides the really small changes, can you please add in support for creating a new group (club)? 

After these revisions, future updates/PR to these files will include:
- PUT and DELETE support  for users and groups
- Testing for groups (If Kris is too busy it might be on you to make a dummy test page) - this is something we should team up on too :)
- Posts (for the groups) reducers/services but again I need to get that table created and Controllers done

**Review by tchen007 — APPROVED** [review]
LGTM - Great work! Woot, this PR can be merged!

**Pull request merged**
